lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -46,21 +47,19 @@ export class PublicClientApplication extends ClientApplication implements IPubli
      */
     constructor(configuration: Configuration) {
         super(configuration);
-        this.initializeBrokering();
     }
 
     /**
      * 
      */
-    private initializeBrokering(): void {
-        if (this.config.system.brokerOptions.actAsBroker) {
+    async initializeBrokering(): Promise<void> {     
+        if (this.config.system.brokerOptions.actAsBroker && !BrowserUtils.isInIframe()) {
             this.broker = new BrokerClientApplication(this.config);
+            this.broker.listenForBrokerMessage();
         } else if (this.config.system.brokerOptions.allowBrokering) {
             this.embeddedApp = new EmbeddedClientApplication(this.config, this.logger, this.browserStorage);
             this.logger.verbose("Acting as child");
-            this.embeddedApp.initiateHandshake().catch((e) => {
-                this.logger.error(`Broker handshake failed: ${e}`);
-            });
+            await this.embeddedApp.initiateHandshake();
```

Comment:
Log messages would be good here, including a warning if both flags are enabled and in they are in the top frame.

Reply:
Done